pkg/binding/bindingpolicy-resolver.go

```diff
@@ -303,6 +316,33 @@ func (resolver *bindingPolicyResolver) ResolutionRequiresSingletonReportedState(
 	return bindingPolicyResolution.requiresSingletonReportedState
 }
 
+func (resolver *bindingPolicyResolver) GetSingletonReportedStateRequestForObject(objId util.ObjectIdentifier) (bool, int) {
+	resolver.RWMutex.RLock()
+	defer resolver.RWMutex.RUnlock()
+	var requested bool
+	for _, resolution := range resolver.bindingPolicyToResolution {
+		matches, thisRequest, _ := resolution.getSingletonReportedStateReturnRequest(objId)
+		if matches && thisRequest {
+			requested = true
+			break
+		}
+	}
+	if !requested {
+		return false, 0
+	}
+	requested = false
+	matchingWECs := sets.New[string]()
+	for _, resolution := range resolver.bindingPolicyToResolution {
```

Comment:
Why not unify the loops?

Reply:
The first loop only computes, does not allocate anything on the heap. For an object that does not request singleton reported state return, the early out saves a lot of heap thrashing. Heap thrashing is expensive, both in terms of "inline" costs and the deferred costs of garbage collection.

Reply:
Fair, perhaps it can be commented in order to quickly be noticed if read in the future?

Reply:
Done